unofficial/c511027120.lua

```diff
@@ -0,0 +1,89 @@
+--爆走特急ロケット・アロー
+--Rocket Arrow Express (Anime)
+--Scripted by the Razgriz
+local s,id=GetID()
+function s.initial_effect(c)
+	c:EnableReviveLimit()
+	--special summon
+	local e1=Effect.CreateEffect(c)
+	e1:SetType(EFFECT_TYPE_FIELD)
+	e1:SetCode(EFFECT_SPSUMMON_PROC)
+	e1:SetProperty(EFFECT_FLAG_UNCOPYABLE+EFFECT_FLAG_CANNOT_DISABLE)
+	e1:SetRange(LOCATION_HAND)
+	e1:SetCondition(s.spcon)
+	c:RegisterEffect(e1)
+	--cannot set
+	local e2=Effect.CreateEffect(c)
+	e2:SetType(EFFECT_TYPE_FIELD)
+	e2:SetCode(EFFECT_CANNOT_MSET)
+	e2:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+	e2:SetRange(LOCATION_MZONE)
+	e2:SetTargetRange(1,0)
+	e2:SetTarget(aux.TRUE)
+	c:RegisterEffect(e2)
+	local e3=e2:Clone()
+	e3:SetCode(EFFECT_CANNOT_SSET)
+	c:RegisterEffect(e3)
+	local e4=e2:Clone()
+	e4:SetCode(EFFECT_CANNOT_TURN_SET)
+	c:RegisterEffect(e4)
+	local e5=e2:Clone()
+	e5:SetCode(EFFECT_CANNOT_SPECIAL_SUMMON)
+	e5:SetTarget(s.sumlimit)
+	c:RegisterEffect(e5)
+	--maintain
+	local e6=Effect.CreateEffect(c)
+	e6:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_CONTINUOUS)
+	e6:SetProperty(EFFECT_FLAG_CANNOT_DISABLE+EFFECT_FLAG_UNCOPYABLE)
+	e6:SetCode(EVENT_PHASE+PHASE_STANDBY)
+	e6:SetRange(LOCATION_MZONE)
+	e6:SetCountLimit(1)
+	e6:SetCondition(s.mtcon)
+	e6:SetOperation(s.mtop)
+	c:RegisterEffect(e6)
+	--spsummon cost
+	local e7=Effect.CreateEffect(c)
+	e7:SetType(EFFECT_TYPE_SINGLE)
+	e7:SetCode(EFFECT_SPSUMMON_COST)
+	e7:SetProperty(EFFECT_FLAG_CANNOT_DISABLE)
+	e7:SetCost(s.spcost)
+	e7:SetOperation(s.spcop)
+	c:RegisterEffect(e7)
+end
+function s.spcon(e,c)
+	if c==nil then return true end
+	return Duel.GetFieldGroupCount(c:GetControler(),LOCATION_ONFIELD,0,nil)==0
+		and Duel.GetLocationCount(c:GetControler(),LOCATION_MZONE)>0
+end
+function s.sumlimit(e,c,sump,sumtype,sumpos,targetp)
+	return (sumpos&POS_FACEDOWN)~=0
+end
+function s.mtcon(e,tp,eg,ep,ev,re,r,rp)
+	return Duel.GetTurnPlayer()==tp
+end
+function s.filter(c) 
+	return c:IsAbleToGraveAsCost()
+end
+function s.mtop(e,tp,eg,ep,ev,re,r,rp)
+	local g=Duel.GetMatchingGroup(s.filter,e:GetHandlerPlayer(),LOCATION_HAND,0,nil)
+	if #g>=5 and Duel.SelectYesNo(tp,aux.Stringid(id,1)) then
+		local hg=aux.SelectUnselectGroup(g,e,tp,5,5,nil,1,tp,HINTMSG_TOGRAVE)

```

Comment:
Don't need SelectUnselectGroup here, it's overkill.
Just a simple g:Select(tp,5,5,nil) should be enough. (and remember the hintselect above)